lib/selection.js

```diff
@@ -0,0 +1,202 @@
+/** A selection of pieces to download.
+ * @typedef {Object} SelectionItem
+ * @property {number} from
+ * @property {number} to
+ * @property {number} offset
+ * @property {number} priority
+ * @property {function} notify
+ * @property {boolean} isStreamSelection
+ */
+
+export class Selection {
+  /**
+   * @type {Array<SelectionItem>}
+   * @private
+   */
+  _items = []
+
+  /**
+   * @public
+   * @param {{from: number, to: number}} item Interval to be removed from the selection
+   * @param {boolean} [isStreamSelection]
+   */
+  remove (item, isStreamSelection = false) {
+    for (let i = 0; i < this._items.length; i++) {
+      const existing = this._items[i]
```

Comment:
you do not check priorities here, remove should remove by priorities!!!!!!!!!!!

Reply:
I added the priority check to the section that compares `isStreamSelection` selections

For regular `select` and `deselect`, priority shouldn't matter:
- If it's an insertion, then it should overwrite whatever was there previously.
- If we are using `deselect`, then the user shouldn't have to know beforehand what the current selection's priority is. Especially when you deselect a range that contains multiple existing ranges - they could all be different selections.